### PR TITLE
Make overlay shortcuts able to be toggled instead of repeatable

### PR DESCRIPTION
**Pull request by ajiiisai**
Currently if you hold any of the shortcuts it just spams the overlay command and personally that's a thing that should be toggable instead of the current behavior.

I don't know if the way I did it is the proper way since I'm not really familiar with the entirety of the codebase.

Let me know if anything isn't as it should.

### CODE REVIEW

**Comment by peppy** [comment]
This seems like fixing with a hammer. Definitely not the right way to go about this.

A starting point would be to find where the actions are being triggered, then addressing the issue at the source.

**Comment by ajiiisai** [comment]
I tried to find where the keybinds were being processed for each overlay but couldn't find it. I assume `osu-framework` is not involved in this behavior and it's something being processed at the `Osu.Game` level right?

**Comment by peppy** [comment]
Have you tried putting a breakpoint in the toggle logic and seeing where it comes from?

**Comment by ajiiisai** [comment]
The `GlobalAction` enum doesn't hold any information about if the action should be repeatable on keydown or not, it's an enum after all so how I should be going around filtering particular GlobalActions that should repeat (`IncreaseVolume`, `DecreaseVolume` etc.)
Or am I understanding how it works wrong?

**Comment by bdach** [comment]
hint 1: the original logic you're looking for is here:

https://github.com/ppy/osu/blob/a2829b4fc7fa7a79e6e884fc9a381a9592eb7aad/osu.Game/Overlays/Toolbar/ToolbarButton.cs#L191-L200

hint 2: `e.Repeat` exists there.

**Comment by ajiiisai** [comment]
I think that should do it, let me know.

**Comment by bdach** [comment]
I've flipped the order of `&&` operands because it read more naturally to me flipped. Other than that I think this is fine and the change makes sense UX-wise, but I'm going to wait for a second approval before going forward with this just to make sure.

Considered adding testing for this but decided against it in the end as it feels like a bit too much.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**
